### PR TITLE
Game_logic

### PR DESCRIPTION
**Pull request by Lichapa**
In this milestone, The following was done:

- create board.rb file in lib directory
- create player.rb file in lib directory
- Add player class to lib/player.rb
- Add board class to lib/board.rb
- Add require_relative to bin./main.rb for lib/player.rb and lib/board.rb

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    display_board
+  end
+
+  # Create a new board
+  def display_board()
+    puts "| #{@space[0]} | #{@space[1]} | #{@space[2]} |"
+    puts '---------------'
+    puts "| #{@space[3]} | #{@space[4]} | #{@space[5]} |"
+    puts '---------------'
+    puts "| #{@space[6]} | #{@space[7]} | #{@space[8]} |"
```
Comment (umairarshadbutt) [inline lib/board.rb:26]: Remove the puts and move into `main.rb`

File: lib/player.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+class Player
+  attr_reader :name, :symbol
+
+  def initialize(name, symbol)
+    @name = name
+    @symbol = symbol
+  end
+
+  def make_move(player, board, move)
+    print "#{name}, It's your turn. Please make your move:"
```
Comment (umairarshadbutt) [inline lib/player.rb]: Remove the print and move into `main.rb`

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @Lichapa 
Such a remarkable project you made :+1:  :tada: 
We, however, do have a few changes to carry out before we proceed (see review comment).

##

Kindly work on the suggested changes 👇.
- [ ] Please remove `puts/print` and `gets.chomp` from `board.rb` and `player.rb` and move them to  `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- [ ] Game is terminating after the draw, the game should ask after the draw if the player wants to play again.
- [ ] Game is giving the message Cat's game! on the first draw. This message is shown when the game is continuously drawing three times. You can explore [Why is a tie in Tic-Tac-Toe called a “Cat's Game?”](https://english.stackexchange.com/questions/155621/why-is-a-tie-in-tic-tac-toe-called-a-cats-game)

##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Review by mahmoud717 — CHANGES_REQUESTED** [review]
### Changes required ♻️ 

**Hello @Lichapa**

Great job with the project so far it looks amazing, but there are still some changes to be made:
 
---
### Requested 🔴 

- [ ] After a player enters his name, you should till him what his token is.

---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

**Review by mahmoud717 — APPROVED** [review]
### STATUS: APPROVED ✔️ ✔️ 

**hi @Lichapa**

**Great job with making all of the requirements. the project is approved, you can go ahead and merge this pull request**

**Pull request merged**